### PR TITLE
Merge master into rt

### PR DESCRIPTION
**Pull request by corlewis**
As usual, it probably only makes sense to review the final post-merge commits. The main impacts from master are making use of the new `arch_requalify` and `arch_global_naming` commands, and the changes to `init_arch_objects`.

### CODE REVIEW

File: spec/machine/RISCV64/MachineOps.thy [hunk 1]
```diff
@@ -84,10 +84,6 @@ consts' timerPrecision :: "64 word"
 consts' max_ticks_to_us :: "64 word"
 consts' max_us_to_ticks :: "64 word"
 
-end
-
-qualify RISCV64 (in Arch)
```
Comment (corlewis) [inline spec/machine/RISCV64/MachineOps.thy:89]: Due to abbreviations inside a `qualify` block seeming not to be properly qualified and then causing problems when being requalified, I moved the following types, abbreviation and definitions into the preceding `Arch` context. This makes all of the qualifications work, but does annoyingly mean that uses of the definitions inside of the `qualify` block need to themselves be qualified. This is the best approach that we were able to find for this.

**Comment by corlewis** [comment]
I believe that the Lint failure can be ignored, they all seem to be from `Requalify_Test.thy`.

**Review by Xaphiosis — APPROVED** [review]
I have reviewed the following commits:
* rt spec proof: update after merge
* rt trivial: rename arch_split -> arch-split again after merge
please let me know if this is sufficient (I don't know of any way to review an actual merge commit)

File: proof/invariant-abstract/Invariants_AI.thy [hunk 2]
```diff
@@ -53,7 +54,9 @@ arch_requalify_consts
   valid_global_vspace_mappings
   pspace_in_kernel_window
 
+  last_machine_time_update
   last_machine_time
+  time_state_update
```
[the remark this answers is not in the document]
Comment (corlewis) [reply proof/invariant-abstract/Invariants_AI.thy:59]: I can't remember if it was just for consistency or if it was needed but I moved these two from a later file, and I mistakenly removed `reply_bits_def` in the merge and found it easier to just add it back here once I realised that it was still needed.

**Comment by corlewis** [comment]
> I have reviewed the following commits:
> 
>     * rt spec proof: update after merge
> 
>     * rt trivial: rename arch_split -> arch-split again after merge
>       please let me know if this is sufficient (I don't know of any way to review an actual merge commit)

That's all I expected to be reviewed, I agree that merge commits themselves can't be reviewed which is why I try to just resolve conflicts inside of it and have any new changes in separate commits.

**Review by michaelmcinerney — APPROVED** [review]
I've still not had any real hands-on experience with the arch-split stuff, but just from reading the documentation and thinking a little about it, this does look good to me. Thanks for doing this merge

**Pull request merged**